uwblueprint/building-up-server
# Email Service for Inviting Users to Team

## Description

Link to ticket: https://www.notion.so/uwblueprintexecs/Email-Service-for-Inviting-Users-to-Team-78326acbe90345c59d15307c48efd4ed

See accompanying client PR: https://github.com/uwblueprint/building-up-client/pull/43

## Approach

If applicable, describe your approach to solving this problem.

- Integrated Twilio's SendGrid API for free (100 emails a day for free) to send general emails. All of the credentials are listed in the ticket as well.
- Added a mutation for retrieving email addresses from client.
- Also added `SENDGRID_API_KEY` to Heroku as config var.

## Testing

- Make sure you add the API key to `.env` for testing
- Pull the branch `sendgrid-service` on building-up-client
- Login to the UI, go to Teams, add your own email address and press send invites. If you add multiple, it should send emails to all of those email addresses.
- You should receive an email in your inbox from hongyichen@uwblueprint.org (temporary email address)
